pydle/connection.py

```diff
@@ -103,8 +101,8 @@ def connected(self):
         return self.reader is not None and self.writer is not None
 
     def stop(self):
-        """ Stop event loop. """
-        self.eventloop.call_soon(self.eventloop.stop)
+        """ Stop connection. """
+        asyncio.run(self.disconnect())
```

Comment:
This will not work correctly, as the event loop is (probably) already running in this context.

`asyncio.run` will create a new event loop.

Although, I question if we still need this method since we should assume pydle's caller owns the event loop.

Reply:
If we assume that pydle's caller owns the event loop, then we cannot call `asyncio.run` here as it will trigger `asyncio.run`'s assertion that there is not a running event loop.

I think we would need to do an `asyncio.create_task(self.disconnect())` here (and hope it doesn't throw an exception)